File: TeddyPet_BE/src/main/java/fpt/teddypet/application/dto/response/RegisterResponse.java
package fpt.teddypet.application.dto.response;

import java.time.LocalDateTime;

/**
 * Response DTO for registration containing resend email cooldown info.
 */
public record RegisterResponse(
        String message,
        long resendCooldownSeconds,
        LocalDateTime canResendAt) {
}

File: teddypet_mobile/lib/data/models/response/auth/register_response.dart
class RegisterResponse {
  final String message;
  final int resendCooldownSeconds;
  final String? canResendAt;

  RegisterResponse({
    required this.message,
    required this.resendCooldownSeconds,
    this.canResendAt,
  });

  factory RegisterResponse.fromJson(Map<String, dynamic> json) {
    return RegisterResponse(
      message: json['message'] ?? '',
      resendCooldownSeconds: json['resendCooldownSeconds'] ?? 0,
      canResendAt: json['canResendAt'],
    );
  }
}
